nerfstudio/process_data/images_to_nerstudio_dataset.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The Nerfstudio Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Processes an image sequence to a nerfstudio compatible dataset."""
+
+from dataclasses import dataclass
+from typing import Optional
+
+from rich.console import Console
+
+from nerfstudio.process_data import equirect_utils, process_data_utils
+from nerfstudio.process_data.colmap_converter_to_nerfstudio_dataset import (
+    ColmapConverterToNerfstudioDataset,
+)
+
+CONSOLE = Console(width=120)
+
+
+@dataclass
+class ImagesToNerfstudioDataset(ColmapConverterToNerfstudioDataset):
+    """Process images into a nerfstudio dataset.
+
+    1. Scales images to a specified size.
+    2. Calculates the camera poses for each image using `COLMAP <https://colmap.github.io/>`_.
+    """
+
+    def main(self) -> None:  # pylint: disable=R0915
```

Comment:
Not needed for this PR, but just wondering @MalcolmMielle do you think there's more refactoring that could be done to consolidate the image / video processing logic?

It'd also be nice to reduce the inheritance layers, but I don't have any concrete suggestions for how to do that cleanly 🤷

Reply:
I've tried my best hahaha. There might be more that can be done later but for now I don't see more :). 

With python 3.10 we could use 'kw_only' on the dataclasses to reduce the inheritance with the "default for order" classes.